rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_publisher.hpp

```diff
@@ -61,8 +61,9 @@ class LifecyclePublisher : public LifecyclePublisherInterface,
     rclcpp::node_interfaces::NodeBaseInterface * node_base,
     const std::string & topic,
     const rclcpp::QoS & qos,
-    const rclcpp::PublisherOptionsWithAllocator<Alloc> & options)
-  : rclcpp::Publisher<MessageT, Alloc>(node_base, topic, qos, options),
+    const rclcpp::PublisherOptionsWithAllocator<Alloc> & options,
+    const rosidl_message_type_support_t & type_support)
```

Comment:
Can we have an overloaded constructor for this to not break with existing code?
The first constructor could get the type support as done above with the macro from `MessageT`